### DISCUSSION TITLE
When clearing a historical task in Airflow, it may incorrectly trigger other tasks with no status to change into running status.

### DISCUSSION THREAD

**Discussion by itsallsame** [apache/airflow#32547] [General]
### Apache Airflow version

2.6.3

### What happened

When clearing a historical task in Airflow, it may incorrectly trigger other tasks with no status to change into running status.
such as i clear task A, airflow will trigger task B。 （Task A and Task B have no dependencies on each other）。

![image](https://github.com/apache/airflow/assets/115528312/bf123308-9162-484f-a019-fbb0213cd55c)

PS：

airflow2.3.1 、2.3.4 also has this problem.

### What you think should happen instead

Do not trigger Task B， just task A is running

### How to reproduce

In airflow WebUI, clear  a historical task.

### Operating System

centos7.6

### Deployment

Virtualenv installation

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
You have not provided a reproducible case. Your DAG, example. logs what's going on are absolutely necessary to be able to assess if what you are saying is an issue with Airflow or your mistake. Until then it is at most a discussion.
